### ISSUE TITLE
API Request is Blocked by Cloudfare (Likely Cloudfare's DDoS Protection) if Made from Github Action

### ISSUE THREAD

**Issue by vnphanquang** [Miodec/monkeytype#2868]
### Did you clear cache before opening an issue?

- [X] I have cleared my cache

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does the issue happen when logged in?

N/A

### Does the issue happen when logged out?

Yes

### Does the issue happen in incognito mode when logged in?

Yes

### Does the issue happen in incognito mode when logged out?

Yes

### Account name

vnphanquang

### Current Behavior

The Public API is all good if request is made from end users themselves (Postman, browser, local dev, ...) but is blocked with a 503 response status ("Service Temporarily Unavailable") by cloudfare if made from github action. The response body is an HTML that renders to this:

![cloudfare_ddos](https://user-images.githubusercontent.com/25895844/164952753-dbabf816-c309-4e32-a1f3-66be9fb7ef67.png)

[Initial discussion on discord can be found here](https://discord.com/channels/713194177403420752/713196019206324306/961226223248351302). I suspect this is Cloudfare DDoS blocking requests from server with high traffic. In my case it was github action server; but I believe some other servers might also be subject to this.

This issue will potential become a blocker for users that rely on the public API running inside a deployed server elsewhere, especially as the apis grow and are used by more and more people.

Please take a look into this. If a workaround is not possible, it should be noted somewhere for others to know about.

Thanks a lot team!

### Expected Behavior

The Public API allows requests from servers such as in github action.

### Steps To Reproduce

1. Setup a github action
2. Call any monkeytype api endpoint in (1)
3. Expect a 503 response

### Environment

Github action

### Anything else?

The github action I have basically does an equivalence to this in a unit test:

```javascript
const response = await fetch(
            'https://api.monkeytype.com/users/personalBests?mode=time&mode2=30',
            {
                headers: {
                    Authorization: 'ApeKey ...',
                },
            }
        );
```

For completion, [this is where the actual code is](https://github.com/vnphanquang/monkeytype-readme/blob/main/src/badge/badge.ts#L15) and [this is one of the github action execuations that failed](https://github.com/vnphanquang/monkeytype-readme/runs/6143851043?check_suite_focus=true#step:9:36)

**Comment by vnphanquang** [comment]
@Bruception sorry to drag you into this. I've talked about this in discord a while back. Just want to let you know I made an issue here for reference.

**Comment by Miodec** [comment]
Ive checked the logs and tried allowing some IP addresses through the firewall. Try now.

**Comment by vnphanquang** [comment]
@Miodec Thanks. Still no luck unfortunately. Have you try lifting firewalls for listed github action IPs? You can [refer to this docs](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-githubs-ip-addresses), which says a `GET` to [https://api.github.com/meta](https://api.github.com/meta) returns a list of said IPs.

From my debugging, the IP of the server my particular action ran on was `20.242.116.228`.

Hope that helps.

**Comment by Miodec** [comment]
> 



> From my debugging, the IP of the server my particular action ran on was `20.242.116.228`.

Hmm, that address in not on the list https://api.github.com/meta

**Comment by vnphanquang** [comment]
Yeah, i'm not sure either. But that ip is likely to change every time the github action runs. I used [@haythem/public-ip action](https://github.com/haythem/public-ip) to find the IP (which essentially queries [https://api.ipify.org](https://api.ipify.org)). I would just stick to the list from github api.

[This stackoverflow thread](https://stackoverflow.com/questions/68011051/how-do-i-whitelist-github-actions-ip-addresses-in-azure-web-app-using-githubs-a) is relevant, if of any help.

**Comment by Miodec** [comment]
Try now

**Comment by Miodec** [comment]
Whats the exact request that youre making? Can you check if `GET https://api.monkeytype.com/` also gets blocked?

**Comment by vnphanquang** [comment]
`GET https://api.monkeytype.com/` is okay. The exact request was `https://api.monkeytype.com/users/personalBests?mode=time&mode2=30`

**Comment by Miodec** [comment]
@vnphanquang Can you test again?

**Comment by vnphanquang** [comment]
> Maybe its getting blocked because youve only been testing on that one endpoint? How long has it been since you last tried? Can you try again? Can you try the personal bests endpoint?

Oh no sincerely apologies i missed the notification for this.

> @vnphanquang Can you test again?

Just tested, it works again. Thanks a lot @Miodec !
